lib/ansible/modules/cloud/amazon/cloudtrail.py

```diff
@@ -595,7 +595,9 @@ def main():
                 pass
             trail = dict()
             trail.update(ct_params)
-            trail['LogFileValidationEnabled'] = ct_params['EnableLogFileValidation']
+            if 'EnableLogFileValidation' not in ct_params:
+                ct_params['EnableLogFileValidation'] = False
+            trail['EnableLogFileValidation'] = ct_params['EnableLogFileValidation']
```

Comment:
This works in China? I thought the param would have to be missing.

Reply:
This isn't making a real request to Cloudtrail in China - it's constructing the spoof return values from a request (hence the fake ARN).

Reply:
Aha. I need more coffee. Thanks for doing this.

Reply:
If I'm reading right, what's happening is that the check mode opens an STS connection a few lines up to make sure you've passed real credentials and it really can connect to the region, but then basically makes up a fake API return output right here... 

```
***********************************
PARSED OUTPUT
{
    "changed": true,
    "exists": false,
    "invocation": {
        "module_args": {
            "aws_access_key": null,
            "aws_secret_key": null,
            "cloudwatch_logs_log_group_arn": null,
            "cloudwatch_logs_role_arn": null,
            "ec2_url": null,
            "enable_log_file_validation": null,
            "enable_logging": true,
            "include_global_events": true,
            "is_multi_region_trail": false,
            "kms_key_id": null,
            "name": "test-trail-1",
            "profile": "illumina-cn-seq-hub-preprod",
            "region": "cn-north-1",
            "s3_bucket_name": "cn-sh-cnn1-cloud-trail",
            "s3_key_prefix": null,
            "security_token": null,
            "sns_topic_name": null,
            "state": "present",
            "tags": {},
            "validate_certs": true
        }
    },
    "trail": {
        "has_custom_event_selectors": false,
        "home_region": "cn-north-1",
        "include_global_service_events": true,
        "is_logging": true,
        "is_multi_region_trail": false,
        "name": "test-trail-1",
        "s3_bucket_name": "cn-sh-cnn1-cloud-trail",
        "tags": {},
        "trail_arn": "arn:aws:cloudtrail:cn-north-1:640673851126:trail/test-trail-1"
    }
}
```

note: "trail_arn": "arn:aws:cloudtrail:cn-north-1:640673851126:trail/test-trail-1" isn't a possible ARN, the partition name would be `aws-cn` for a real trail.

Reply:
Ha, no problem. Thanks for picking up bug reports that affect a tiiiiiiny percentage of Cloudtrail users :)